src/vivarium/framework/time.py

```diff
@@ -12,53 +12,102 @@
-        assert self._time is not None, "No start time provided"
-        return self._time
+        if not self._clock_time:
+            raise ValueError("No start time provided")
+        return self._clock_time
 
     @property
     def stop_time(self) -> Time:
         """The time at which the simulation will stop."""
-        assert self._stop_time is not None, "No stop time provided"
+        if not self._stop_time:
+            raise ValueError("No stop time provided")
         return self._stop_time
 
     @property
     def step_size(self) -> Timedelta:
         """The size of the next time step."""
-        assert self._step_size is not None, "No step size provided"
-        return self._step_size
-
-    def step_forward(self) -> None:
-        """Advances the clock by the current step size."""
-        self._time += self.step_size
+        if not self._clock_step_size:
+            raise ValueError("No step size provided")
+        return self._clock_step_size
+
+    def simulant_next_event_times(self, index: pd.Index) -> pd.Series:
+        """The next time each simulant will be updated."""
+        if not self.population_view:
+            raise ValueError("No population view defined")
+        return self.population_view.subview(["next_event_time"]).get(index)
+
+    def simulant_step_sizes(self, index: pd.Index) -> pd.Series:
+        """The step size for each simulant."""
+        if not self.population_view:
+            raise ValueError("No population view defined")
+        return self.population_view.subview(["step_size"]).get(index)
+
+    def setup(self, builder: "Builder"):
+        builder.population.initializes_simulants(
+            self.on_initialize_simulants, creates_columns=self.columns_created
+        )
+        self.population_view = builder.population.get_view(columns=self.columns_created)
+
+    def on_initialize_simulants(self, pop_data):
+        """Sets the next_event_time and step_size columns for each simulant"""
+        simulant_clocks = pd.DataFrame(
+            {
+                "next_event_time": [self.time + self.step_size] * len(pop_data.index),
+                "step_size": [self.step_size] * len(pop_data.index),
+            },
+            index=pop_data.index,
+        )
+        self.population_view.update(simulant_clocks)
 
-    def step_backward(self):
+    def step_backward(self) -> None:
         """Rewinds the clock by the current step size."""
-        self._time -= self.step_size
+        self._clock_time -= self.step_size
+
+    def step_forward(self, index: pd.Index) -> None:
+        """Advances the clock by the current step size, and updates aligned simulant clocks."""
+        event_time = self.time + self.step_size
+        pop_to_update = self.aligned_pop(index, event_time)
+        pop_to_update["next_event_time"] = event_time + pop_to_update["step_size"]
+        self.population_view.update(pop_to_update)
+        self._clock_time += self.step_size
```

Comment:
I thought about that, but given all the conceptual difficulties around event times and clock times I thought it might be worth a couple lines to be very explicit in keeping the clock and the "event time" separate. I don't have strong feelings about it, though